internal/state/dataplane/configuration.go

```diff
@@ -70,14 +73,41 @@ type PathRule struct {
 	MatchRules []MatchRule
 }
 
+type Map struct {
```

Comment:
in the dataplane package, we have a data plane configuration which is independent from NGINX. This allow us to either swap how we implement something in NGINX (for example, use maps instead of njs for request routing) or swap or integrate data planes in the future. considering that, including the Map, which is NGINX specific, goes against the goals of the data plane package.
a good related examples - how we implement split clients https://github.com/nginxinc/nginx-kubernetes-gateway/blob/dbcef59106ad1a6dc63963863ebc154a7a5d6daa/internal/nginx/config/split_clients.go#L20  . SplintClients is NGINX specific directive and it is not part of the data plane package.